tests/Feature/ReviewTest.php

```diff
@@ -0,0 +1,89 @@
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+
+class ReviewTest extends TestCase
+{
+
+    use DatabaseMigrations;
+
+    public function test_add_book_review(): void
+    {
+        Review::factory()->create();
+
+        $response = $this->postJson('/api/reviews', ['name' => 'Book', 'rating' => '4', 'review' => 'What an amazing book!', 'book_id' => '20']);
+
+        $response->assertStatus(201)
+            ->assertJson([
+                'message' => 'Review created'
+            ]);
+    }
+
+    public function test_add_book_review_name_invalid(): void
+    {
+        $response = $this->postJson('/api/reviews', ['name' => '', 'rating' => '4', 'review' => 'What an amazing book!', 'book_id' => '20']);
+
+        $response->assertStatus(422)
+            ->assertJson([
+                'message' => 'The name field is required.'
+            ]);
+    }
+
+    public function test_add_book_review_rating_invalid(): void
+    {
+        $response = $this->postJson('/api/reviews', ['name' => 'Book', 'rating' => '', 'review' => 'What an amazing book!', 'book_id' => '20']);
+
+        $response->assertStatus(422)
+            ->assertJson([
+                'message' => 'The rating field is required.'
+            ]);
+    }
+
+    public function test_add_book_review_review_invalid(): void
+    {
+        $response = $this->postJson('/api/reviews', ['name' => 'Book', 'rating' => '4', 'review' => '', 'book_id' => '20']);
+
+        $response->assertStatus(422)
+            ->assertJson([
+                'message' => 'The review field is required.'
+            ]);
+    }
+
+    public function test_add_book_review_book_id_invalid(): void
+    {
+        $response = $this->postJson('/api/reviews', ['name' => 'Book', 'rating' => '4', 'review' => 'Nice book.', 'book_id' => '']);
+
+        $response->assertStatus(422)
+            ->assertJson([
+                'message' => 'The book id field is required.'
+            ]);
+    }
+
+    public function test_add_book_review_rating_outside_min(): void
+    {
+        $response = $this->postJson('/api/reviews', ['name' => 'Book', 'rating' => '-1', 'review' => 'What an amazing book!', 'book_id' => '20']);
+
+        $response->assertStatus(422)
+            ->assertJson([
+                'message' => 'The rating field is required.',
+                'message' => 'The rating field must be at least 0.',
+            ]);
+    }
+
+    public function test_add_book_review_rating_outside_max(): void
+    {
+        $response = $this->postJson('/api/reviews', ['name' => 'Book', 'rating' => '6', 'review' => 'What an amazing book!', 'book_id' => '20']);
+
+        $response->assertStatus(422)
+            ->assertJson([
+                'message' => 'The rating field is required.',
+                'message' => 'The rating field must not be greater than 5.',
+            ]);
+    }
+}
```

Comment:
When you are testing for validation failures, it's better to use `assertInvalid` instead of `assertJson` - it does all the work for you

https://laravel.com/docs/10.x/http-tests#validation-assert-invalid

Reply:
It also automatically checks for a 422 status code, so need to assertStatus either